core/controllers/editor.py

```diff
@@ -739,12 +748,14 @@ def put(self, exploration_id):
         exp_user_data = exp_services.get_user_exploration_data(
             self.user_id, exploration_id)
         # If the draft_change_list_id is False, have the user discard the draft
-        # changes. We save the draft to the datastore even if the version is
-        # invalid, so that it is available for recovery later.
+        # changes. We save the draft to the datastore even if the changes are
+        # not mergeable, so that it is available for recovery later.
```

Comment:
I would suggest updating the draft_version at AutoSaveHandler if the changes_are_mergable. I'll approve the PR but let's create an issue and discuss it further. (This comment should not block the PR as the overall user journey is working fine!)

Reply:
Yeah, I guess there is some confusion regarding this. I'll create an issue and there we'll discuss and proceed further. Thanks!